docs/en/stores/supplemental.md

```diff
@@ -0,0 +1,425 @@
+
+> Using Immutable
+
+```ts
+import State from './interfaces';
+import Store from '@dojo/framework/stores/Store';
+import Registry from '@dojo/framework/widget-core/Registry';
+import ImmutableState from '@dojo/framework/stores/state/ImmutableState';
+
+const registry = new Registry();
+const customStore = new ImmutableState<State>();
+const store = new Store<State>({ store: customStore });
+```
+
+## Local Storage
+
+Dojo Stores provides a collection of tools to leverage local storage.
+
+The local storage middleware watches specified paths for changes and stores them locally on disk using the `id` provided to the `collector` and structure as defined by the path.
+
+> Local storage middleware
+
+```ts
+export const myProcess = createProcess(
+	'my-process',
+	[command],
+	collector('my-process', (path) => {
+		return [path('state', 'to', 'save'), path('other', 'state', 'to', 'save')];
+	})
+);
+```
+
+The `load` function hydrates a store from `LocalStorage`
+
+> Hydrating state
+
+```ts
+import { load } from '@dojo/framework/stores/middleware/localStorage';
+import { Store } from '@dojo/framework/stores/Store';
+
+const store = new Store();
+load('my-process', store);
+```
+
+Note that data is serialized for storage and the data is overwritten after each process call. This implementation is not appropriate for non-serializable data (e.g. `Date` and `ArrayBuffer`).
+
+# Subscribing to Store changes
+
+The `Store` has an `onChange(path, callback)` method that takes a path or an array of paths and will call a callback function whenever that state changes.
+
+> main.ts
+
+```ts
+const store = new Store<State>();
+const { path } = store;
+
+store.onChange(path('auth', 'token'), () => {
+	console.log('new login');
+});
+
+store.onChange([path('users', 'current'), path('users', 'list')], () => {
+	// Make sure the current user is in the user list
+});
+```
+
+The `Store` also has an `invalidate` event that fires any time the store changes.
+
+> main.ts
+
+```ts
+store.on('invalidate', () => {
+	// do something when the store's state has been updated.
+});
+```
+
+# JSON Patch and Store operations
+
+In the basic usage guide we show how Dojo Stores use operations in detail to make changes to the underlying state of an application. There are a few more advanced points that are not covered in the basic usage guide.
+
+First and foremost Dojo Stores was designed with simplicity in mind. For instance, operations will automatically create the underlying structure necessary to support an `add` or `replace` operation.
+
+> Deep `add` in an uninitialized store
+
+```ts
+import Store from '@dojo/framework/stores/Store';
+import { add } from '@dojo/framework/stores/state/operations';
+
+const store = new Store<State>();
+const { at, path, apply } = store;
+const user = { id: '0', name: 'Paul' };
+
+apply([add(at(path('users', 'list'), 10), user)]);
+```
+
+> Result
+
+```json
```

Comment:
Most of our other guide examples use spaces for indentation, so we may want to change this for consistency?